ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java

```diff
@@ -0,0 +1,174 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.util;
+
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.confluent.kafka.schemaregistry.client.SchemaMetadata;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.ksql.ddl.DdlConfig;
+import io.confluent.ksql.exception.ExceptionUtil;
+import io.confluent.ksql.metastore.MetastoreUtil;
+import io.confluent.ksql.parser.SqlFormatter;
+import io.confluent.ksql.parser.tree.AbstractStreamCreateStatement;
+import io.confluent.ksql.parser.tree.CreateStream;
+import io.confluent.ksql.parser.tree.CreateTable;
+import io.confluent.ksql.parser.tree.Expression;
+import io.confluent.ksql.parser.tree.StringLiteral;
+import io.confluent.ksql.parser.tree.TableElement;
+import io.confluent.ksql.serde.DataSource;
+
+public class AvroUtil {
+
+  private static final Logger log = LoggerFactory.getLogger(AvroUtil.class);
+
+  public static synchronized Pair<AbstractStreamCreateStatement, String> checkAndSetAvroSchema(AbstractStreamCreateStatement
+           abstractStreamCreateStatement,
+       Map<String, Object> streamsProperties, SchemaRegistryClient schemaRegistryClient) {
+    Map<String,Expression> ddlProperties = abstractStreamCreateStatement.getProperties();
+    if (!ddlProperties.containsKey(DdlConfig.VALUE_FORMAT_PROPERTY)) {
+      throw
+          new KsqlException(String.format("%s should be set in WITH clause of CREATE "
+                                          + "STRAEM/TABLE statement.", DdlConfig.VALUE_FORMAT_PROPERTY));
+    }
+    final String serde = StringUtil.cleanQuotes(
+        ddlProperties.get(DdlConfig.VALUE_FORMAT_PROPERTY).toString());
+    if (!serde.equalsIgnoreCase(DataSource.AVRO_SERDE_NAME)) {
+      return new Pair<>(abstractStreamCreateStatement, null);
+    }
+
+    String kafkaTopicName = StringUtil.cleanQuotes(
+        ddlProperties.get(DdlConfig.KAFKA_TOPIC_NAME_PROPERTY).toString());
+    try {
+      SchemaMetadata schemaMetadata;
+      if (abstractStreamCreateStatement.getProperties().containsKey(KsqlConstants.AVRO_SCHEMA_ID)) {
+        int schemaId = -1;
```

Comment:
initializing the variable is redundant as an exception is thrown if it fails to parse the schema id